src/ESPEasy.ino

```diff
@@ -1015,14 +1008,16 @@ void backgroundtasks()
         serial();
       }
     }
-    if (wifiConnected) {
+    if (webserverRunning) {
       WebServer.handleClient();
+    }
+    if (WiFi.getMode() != WIFI_OFF) {
       checkUDP();
     }
   }
 
   // process DNS, only used if the ESP has no valid WiFi config
-  if (dnsServerActive && wifiConnected)
+  if (dnsServerActive)
```

Comment:
@TD-er all sort of wifi stuff should go to ESPEasyWifi.ino, what do you think?

Reply:
Moving code around to the more logical locations was my next step indeed, but I wanted to do it in steps/commits, so it is still traceable.
But the one line you marked is just one that is not WiFi related and was for sure a bug that took me a while to pinpoint.
It was the source of reboots at initial WiFi setup and probably also causing crashes when there was no WiFi connection, but only the AP active.
What was happening here, is that the dns server (and web server) kept on adding stuff to their buffers unless they were being attended.
So you would run out of memory in minutes (or seconds if somehow requests came in)

I wouldn't be surprised if this also lead to crashes of nodes when they were trying to reconnect to WiFi.

Reply:
nice catch :)